src/core/thread/mesh_forwarder.cpp

```diff
@@ -2113,4 +2116,19 @@ void MeshForwarder::HandleDataRequest(const Mac::Address &aMacSource, const Thre
     {}
 }
 
+void MeshForwarder::HandleDataPollTimeout(void *aContext)
+{
+    static_cast<MeshForwarder *>(aContext)->HandleDataPollTimeout();
+}
+
+void MeshForwarder::HandleDataPollTimeout(void)
+{
+    mBacktoBackPollTimeoutCounter++;
+
+    if (mBacktoBackPollTimeoutCounter < kMaxBacktoBackPollTimeouts)
+    {
+        SendMacDataRequest();
+    }
```

Comment:
I agree... It's a good idea to set the counter back to zero once it hits the max. It makes it go through the 5 polls the next time polling starts....Will change it.